### PR TITLE
added browser test to verify attribution section is collapsed by default and expands on clicking

### PR DESCRIPTION
**Pull request by Malar-Natarajan**
GIVEN: A page URL in the format of {base_url}/books/{book_slug}/pages/{page_slug}
    WHEN: The page is fully loaded
    THEN: The attribution section is collapsed by default
    AND The attribution section opens when clicked

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -71,3 +71,6 @@ pip-delete-this-directory.txt
 
 # pytest report
 ui-test.html
+pytest-selenium/tests/test_attribution_pages.py
+pytest-selenium/tests/test_attribution_pom.py
```
Comment (TomWoodward) [inline .gitignore]: why are these here?

File: .gitignore [hunk 2]
```diff
@@ -71,3 +71,6 @@ pip-delete-this-directory.txt
...
+.idea/
```
Comment (TomWoodward) [inline .gitignore]: please use your [global gitignore](https://help.github.com/en/articles/ignoring-files#create-a-global-gitignore) for editor files

**Review by TomWoodward — CHANGES_REQUESTED** [review]

File: pytest-selenium/tests/test_attribution_status.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+@markers.test_case("C476303")
+@markers.parametrize("book_slug,page_slug", [("college-physics", "preface")])
+@markers.nondestructive
+def test_attribution_collapsed_by_default_expands_when_clicked(
+    selenium, base_url, book_slug, page_slug
+):
+
+    """
+    attribution section is initially collapsed, expands when clicked
```
Comment (m1yag1) [inline pytest-selenium/tests/test_attribution_status.py]: Based on the style guide the first line should have the short description.

Comment (m1yag1) [reply pytest-selenium/tests/test_attribution_status.py]: https://wiki.mozilla.org/QA/Execution/Web_Testing/Docs/Automation/StyleGuide

File: pytest-selenium/tests/test_attribution_status.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+    GIVEN: A page URL in the format of {base_url}/books/{book_slug}/pages/{page_slug}
```
Comment (m1yag1) [inline pytest-selenium/tests/test_attribution_status.py]: Are you moving the Given When Then into the docstring?

File: pytest-selenium/tests/test_attribution_status.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    attribution = content.attribution
+
+    assert not attribution.is_open
+
+    attribution.click_attribution_link()
+
+    assert attribution.is_open
+
+    attribution.click_attribution_link()
+    # """click on attribution again and verify it closes
```
Comment (m1yag1) [inline pytest-selenium/tests/test_attribution_status.py]: You are using a docstring here which should only come after a `def` or `class` definition. comments within code use a `# `

File: pytest-selenium/tests/test_attribution_status.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    selenium, base_url, book_slug, page_slug
+):
+
+    """
+    attribution section is initially collapsed, expands when clicked
+
+    GIVEN: A page URL in the format of {base_url}/books/{book_slug}/pages/{page_slug}
+    WHEN: The page is fully loaded
+    THEN: The attribution section is collapsed by default
+    AND The attribution section opens when clicked
```
Comment (m1yag1) [inline pytest-selenium/tests/test_attribution_status.py]: Also, not a consistent use of the `:` after the test section designator. should be `AND: `

File: .gitignore (hunk 2 above)
Comment (Malar-Natarajan) [reply .gitignore]: @TomWoodward I removed these from the .gitignore and committed the changes.

File: pytest-selenium/tests/test_attribution_status.py (hunk 3 above)
Comment (Malar-Natarajan) [reply pytest-selenium/tests/test_attribution_status.py]: @m1yag1 I updated the comments and docstrings

File: pytest-selenium/tests/test_attribution_status.py (hunk 5 above)
Comment (Malar-Natarajan) [reply pytest-selenium/tests/test_attribution_status.py]: @m1yag1  I updated the comments and docstrings

File: pytest-selenium/tests/test_attribution_status.py (hunk 6 above)
Comment (Malar-Natarajan) [reply pytest-selenium/tests/test_attribution_status.py]: Updated

File: pytest-selenium/tests/test_attribution_status.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+from tests import markers
+
+
+@markers.test_case("C476303")
+@markers.parametrize("book_slug,page_slug", [("college-physics", "preface")])
+@markers.nondestructive
+def test_attribution_collapsed_by_default_expands_when_clicked(
+    selenium, base_url, book_slug, page_slug
+):
+
```
Comment (m1yag1) [inline pytest-selenium/tests/test_attribution_status.py]: No line break here. Capitalize first word in docstring. `Attribution`

Comment (Malar-Natarajan) [reply pytest-selenium/tests/test_attribution_status.py]: @m1yag1 Updated

File: pytest-selenium/tests/test_attribution_status.py (hunk 5 above)
Comment (m1yag1) [reply pytest-selenium/tests/test_attribution_status.py]: What's up with line 26. I still see a comment with `"""` inside the code

Comment (Malar-Natarajan) [reply pytest-selenium/tests/test_attribution_status.py]: @m1yag1  Thats my bad....I updated this

File: pytest-selenium/tests/test_attribution_status.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    # THEN: The attribution section is collapsed by default
+    assert not attribution.is_open
+
+    # AND: The attribution section opens when clicked
+    attribution.click_attribution_link()
+    assert attribution.is_open
+
+    attribution.click_attribution_link()
+    """click on attribution again and verify it closes"""
```
Comment (TomWoodward) [inline pytest-selenium/tests/test_attribution_status.py]: shouldn't this be a `#` comment?

Comment (Malar-Natarajan) [reply pytest-selenium/tests/test_attribution_status.py]: @TomWoodward 
I remember Mike saying, I cannot place a comment above a method/function, instead use docstring 
@m1yag1 is my memory bad?

Comment (m1yag1) [reply pytest-selenium/tests/test_attribution_status.py]: Yea, that's the right use of a docstring but I think naming the test appropriately would remove the need for the docstring.

Comment (Malar-Natarajan) [reply pytest-selenium/tests/test_attribution_status.py]: Ok. So I removed the docstring for the click function and just have a comment to explain the assertion

**Review by m1yag1 — DISMISSED** [review]

File: pytest-selenium/pages/content.py [hunk 9]
```diff
@@ -108,4 +104,4 @@ def section_url(self):
 
         def click_attribution_link(self):
             self.offscreen_click(self.attribution_link)
-            return self.page.attribution.wait_for_region_to_display()
+            self.page.attribution.wait_for_region_to_display()
```
Comment (m1yag1) [inline pytest-selenium/pages/content.py]: Why was this return removed? It should return the page so that we have a reference to it in the code.

Comment (Malar-Natarajan) [reply pytest-selenium/pages/content.py]: @m1yag1 
I was discussing about some random locators with Greg and he suggested that a return is not necessary since we are already in that page

Comment (m1yag1) [reply pytest-selenium/pages/content.py]: ok then place a `return` only. you don't need to put anything after.

Comment (Malar-Natarajan) [reply pytest-selenium/pages/content.py]: @m1yag1 
do you mean just have the click function and remove the whole wait for region code
         def click_attribution_link(self):
             self.offscreen_click(self.attribution_link)

Comment (Malar-Natarajan) [reply pytest-selenium/pages/content.py]: Mike resolving this conservation to leave the code as such

File: pytest-selenium/tests/test_attribution.py [hunk 10]
```diff
@@ -23,5 +23,30 @@ def test_section_url_in_citation_text_shows_url_for_current_page(
...
+    assert not attribution.is_open
+
+    # AND: The attribution section opens when clicked
+    attribution.click_attribution_link()
+    assert attribution.is_open
+
+    attribution.click_attribution_link()
+
+    # AND: clicking on attribution again collapses it
+    assert not attribution.is_open
```
Comment (TomWoodward) [inline pytest-selenium/tests/test_attribution.py]: this formatting is inconsistent, i would expect it to be 

```python
    # AND: The attribution section opens when clicked
    attribution.click_attribution_link()
    assert attribution.is_open

    # AND: clicking on attribution again collapses it
    attribution.click_attribution_link()
    assert not attribution.is_open
```

**Review by TomWoodward — CHANGES_REQUESTED** [review]

**Review by TomWoodward — APPROVED** [review]
